### ISSUE TITLE
Monocle 3 - orderCells not working

### ISSUE THREAD

**Issue by ksr2018** [cole-trapnell-lab/monocle-release#258]
Hi all, 
I have a large dataset with ~102,000 cells. I used the Monocle 3 workflow: preprocessCDS -> reduceDimension via UMAP -> clusterCells -> learnGraph with DDRTree -> orderCells
The clustering looks good and is consistent with what I get from Seurat.

But I get these warning messages from learnGraph:
`Warning messages:
1: Deprecated, use tibble::rownames_to_column() instead. 
2: Deprecated, use tibble::rownames_to_column() instead. 
3: Deprecated, use tibble::rownames_to_column() instead. 
4: Deprecated, use tibble::rownames_to_column() instead. 
5: Deprecated, use tibble::rownames_to_column() instead. 
6: Deprecated, use tibble::rownames_to_column() instead. 
7: Deprecated, use tibble::rownames_to_column() instead.`

I tried orderCells but it doesn't seem to be doing anything. I selected some nodes from the interactive viewer or I used get_correct_root_state to define cells I want to set as root, but each time I get the following verbose message:

`CellDataSet (storageMode: environment)
assayData: 37991 features, 102551 samples 
  element names: exprs 
protocolData: none
phenoData
  sampleNames: AAACCTGAGACTGGGT-1 AAACCTGAGAGCTATA-1 ... TTTGTCATCTTATCTG-6 (102551 total)
  varLabels: nGene nUMI ... Pseudotime (9 total)
  varMetadata: labelDescription
featureData
  featureNames: Gm16088 U6 ... Gm3395 (37991 total)
  fvarLabels: gene_short_name use_for_ordering
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
Annotation: `

Next I ran plot_cell_trajectory with color_by = "Pseudotime", and it returns:

`Error in `[.data.frame`(pData(cds), , color_by) : 
  undefined columns selected`

Any idea what I'm doing wrong?

**Comment by dwucsf** [comment]
Getting this same error message. Did you happen to figure it out?

**Comment by ksr2018** [comment]
@dwucsf I have not gotten past these errors. Have you had any luck?

**Comment by dwucsf** [comment]
Yeah I got past it. Can you do pData(cds) to make sure you have a Pseudotime column added after you ran orderCells(cds)?

**Comment by ksr2018** [comment]
@dwucsf, there is no Pseudotime column in my pData after I run orderCells.

**Comment by dwucsf** [comment]
Ok that's your problem. You need pseudotime values to run plot_cell_trajectory. The pseudotime value column is added after you run orderCells(cds). Did orderCells(cds) run successfully?

**Comment by ksr2018** [comment]
I don't think orderCells worked. I selected some nodes from the interactive viewer or I used get_correct_root_state, but I got the following verbose message:

CellDataSet (storageMode: environment) assayData: 37991 features, 102551 samples element names: exprs protocolData: none phenoData sampleNames: AAACCTGAGACTGGGT-1 AAACCTGAGAGCTATA-1 ... TTTGTCATCTTATCTG-6 (102551 total) varLabels: nGene nUMI ... Pseudotime (9 total) varMetadata: labelDescription featureData featureNames: Gm16088 U6 ... Gm3395 (37991 total) fvarLabels: gene_short_name use_for_ordering fvarMetadata: labelDescription experimentData: use 'experimentData(object)' Annotation:

**Comment by dwucsf** [comment]
I don't have experience with the get_correct_root_state function but have used the interactive viewer to select ONE node. Do you select multiple nodes? Do you hit "Finish" after you select the node?

**Comment by ksr2018** [comment]
I have tried selecting either one or more than one nodes and clicked Finish. But still no Pseudotime column in pData

**Comment by dwucsf** [comment]
Hey @ksr2018, sorry for the delay in following up. Did you end up figuring it out? If not, I can continue trying to help troubleshoot with you

**Comment by ksr2018** [comment]
Hi @dwucsf , thanks for taking the time to help me. I have not figured it out. When you run orderCells, do you also get a message similar to this?
`CellDataSet (storageMode: environment) assayData: 37991 features, 102551 samples element names: exprs protocolData: none phenoData sampleNames: AAACCTGAGACTGGGT-1 AAACCTGAGAGCTATA-1 ... TTTGTCATCTTATCTG-6 (102551 total) varLabels: nGene nUMI ... Pseudotime (9 total) varMetadata: labelDescription featureData featureNames: Gm16088 U6 ... Gm3395 (37991 total) fvarLabels: gene_short_name use_for_ordering fvarMetadata: labelDescription experimentData: use 'experimentData(object)' Annotation:`

**Comment by dwucsf** [comment]
Sorry again for the late reply. Can you try the plot_cell_trajectory function after you do the LearnGraph step? I want to see if you actually have a trajectory constructed. If you don't, that means the problem lies in the Learngraph step or before. 

plot_cell_trajectory(cds,
                     color_by = "cell_type2") +
                     scale_color_manual(values = cell_type_color)
